integration/README.md

```diff
@@ -0,0 +1,37 @@
+# Integration tests
+
+This directory contains integration tests for the project.
+
+The tree structure generally mirrors the source code tree structure.
+
+Requirements for new files in this directory:
+* Every integration test file **must** use the `integration` build tag
+* Every package **must** be named after its directory with `_integration` appended
+  * Requiring a different package name for integration tests avoids aliasing with the main package.
+
+These requirements are enforced by a unit test in this directory.
```

Comment:
> These requirements are enforced by a unit test in this directory.

nit: Sounds more like a custom linter's job, can be done https://golangci-lint.run/plugins/module-plugins/  https://golangci-lint.run/plugins/go-plugins/ (haven't tried myself, so it's not really a recommendation, just FYI).

Reply:
Interesting, yeah sounds like a good fit.